glide-core/redis-rs/redis/src/cluster_async/pipeline_routing.rs

```diff
@@ -467,43 +523,552 @@ pub fn process_pipeline_responses(
+                .await
+                {
+                    Ok((new_responses, new_addresses_and_indices)) => {
+                        // Update responses for the retried commands
+                        responses = new_responses;
+                        addresses_and_indices = new_addresses_and_indices;
+                    }
+                    Err(e) => {
+                        break Err(e); // If we get into here, it's because `add_pipeline_result` failed to find the matching index
+                    }
+                }
+            }
+
+            // If we get into here, it's because `add_pipeline_result` failed to find the matching index
+            Err(e) => break Err(e),
+        }
+    }
+}
+
+/// Handles the retry logic for a given retry map.
+///
+/// This function processes the retry map, which contains commands that need to be retried based on their retry method.
+/// It handles the different retry methods such as reconnecting, waiting and retrying, or redirecting commands to the appropriate nodes.
+///
+/// # Arguments
+///
+/// * `retry_map` - A map where the key is the `RetryMethod` and the value is a vector of tuples containing:
+///   - The index and optional inner index of the command in the pipeline.
+///   - The address of the node where the command was originally sent.
+///   - The `ServerError` that occurred.
+/// * `core` - The core object that provides access to connection locks and other resources.
+/// * `pipeline` - A reference to the original pipeline containing the commands.
+/// * `retry` - The retry counter.
+/// * `pipeline_responses` - A mutable reference to the collection of pipeline responses.
+/// * `response_policies` - A list of routing info and response policies to the pipeline commands.
+///
+/// # Returns
+///
+/// A `Result` containing a tuple:
+/// - A vector of results for each sub-pipeline execution.
+/// - A vector of (address, indices) pairs indicating where each response should be placed.
+pub async fn handle_retry_map<C>(
+    retry_map: RetryMap,
+    core: Core<C>,
+    pipeline: &crate::Pipeline,
+    retry: u32,
+    pipeline_responses: &mut PipelineResponses,
+    response_policies: &mut Vec<(usize, MultipleNodeRoutingInfo, Option<ResponsePolicy>)>,
+) -> Result<
+    (
+        Vec<Result<RedisResult<Response>, RecvError>>,
+        Vec<(String, Vec<(usize, Option<usize>)>)>,
+    ),
+    (OperationTarget, RedisError),
+>
+where
+    C: Clone + ConnectionLike + Connect + Send + Sync + 'static,
+{
+    // Create a new pipeline map to map the retried commands to their corresponding node.
+    let mut pipeline_map = NodePipelineMap::new();
+    for (retry_method, indices_addresses_and_error) in retry_map {
+        match retry_method {
+            RetryMethod::NoRetry => {
+                // The server error was already added to the pipeline responses, so we can just continue.
+            }
+            RetryMethod::Reconnect | RetryMethod::ReconnectAndRetry => {
```

Comment:
when we need to reconnect without retrying, use trigger_refresh_connection_tasks